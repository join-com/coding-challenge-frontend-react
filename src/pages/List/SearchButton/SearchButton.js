
import React from 'react'
import styled from 'styled-components'


const Button = styled.button`
  background-color: 'blue';
`
export default function SearchButton({onClick}) {
  return <Button onClick={onClick}>Find cases</Button>
}
