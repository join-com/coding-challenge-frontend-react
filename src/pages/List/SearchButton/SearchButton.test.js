import React from 'react'
import { mount, shallow } from 'enzyme'

import SearchButton from './SearchButton'

it('renders the Button without crashing', () => {
  const wrapper = shallow(<SearchButton />)
  expect(wrapper).toMatchSnapshot()
})

it('triggers onClick when clicked', () => {
  const spy = jest.fn()
  const wrapper = mount(<SearchButton onClick={spy} />)
  wrapper.simulate('click')
  expect(spy).toHaveBeenCalledTimes(1)
})
