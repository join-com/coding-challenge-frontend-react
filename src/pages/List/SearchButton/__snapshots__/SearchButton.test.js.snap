// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Button without crashing 1`] = `
<styled.button>
  Find cases
</styled.button>
`;
